### DISCUSSION TITLE
Any medic stuff in the pipeline?

### DISCUSSION THREAD

**Discussion by Zzzzz124** [Fedoraware/Fedoraware#985] [Ideas]
This is god tier hacks for aimbot and other classes, but i main medic and so far medic stuff seems unfinished, I wish the auto vaccinator would work as other hacks, as well as something like auto heal the lowest team mates (on bind would be awesome).

Rest is amazing, thanks all the devs!

**Comment by SeasonalFerret** [comment]
I said it before I'll say it again, nullcore before the shitty rewrite had the best auto-vaccinator ever. It had settings to prioritize you or your healing partner, actually predicted if rockets would hit you instead of just "if within Xhu", could quickly flash everyone in a radius if a crit rocket was coming (but only those that would actually die from it (also did this on regular uber)), would spam all bubbles if you were about to die just so your healing target could get all resists even after you die, etc.
Even the auto-kritzkrieg had logic that would pop kritz on your healing target if _they_ saw at least 5 enemies. 
Had danger prediction for quickfix/uber based on how many enemies were nearby, auto-pop uber before a backstab, etc.

Then V9 decided he wanted to make nullcore slimmer by getting rid of all of the features... ffs.

**Comment by johnathon-walnut** [comment]
@SylveonBottle

**Comment by johnathon-walnut** [comment]
Oops

**Comment by johnathon-walnut** [comment]
@SylveonBottle 

Write me a bullet point list of everything you'd like auto vaccinator to do and there is a 1/50 chance I will look into it when I get home tonight

**Comment by SeasonalFerret** [comment]
I've written everything here: https://hastebin.com/pemulewena.sql

**Comment by johnathon-walnut** [comment]
https://user-images.githubusercontent.com/36936588/204005486-35229dc4-2d25-4c19-938a-64eeb29af4ca.mp4

**Comment by SeasonalFerret** [comment]
The general idea is that we should switch to whatever benefits us the most, and pop whatever might kill us if we don't pop.

We don't need to worry about sticky bombs if their owner is nowhere in sight (since there's little chance for the demo to set them off).
We don't need to worry about pyros if the target is a heavy at full hp (since he can just kill them).
We always need to pop against snipers if they're scoped in on us/target and can kill in one shot.
Etc.

I forgot to add, but if there's no active threats at all and we/target is on fire, we should switch to Fire to mitigate afterburn. Even popping it if we're about to die from it. Lmaobox never got this right, and it ignores afterburn, even if we're about to die from it.

Settings to customize when to pop against snipers, panic flashing, etc are all optional. We biggest prioritize are keeping us/target safe from snipers, PREDICTING rockets, popping on things that'll kill us (rocket + low hp, crit rocket, etc), etc.

**Comment by SeasonalFerret** [comment]
Keep in mind that everything I wrote, nullcore already had before the rewrite. And more. Nullcore had that whole "danger prediction" system that gave a score to all enemy players, and could pop uber/kritz depending on how many dangerous players were within view.

**Comment by johnathon-walnut** [comment]
> Keep in mind that everything I wrote, nullcore already had before the rewrite. And more. Nullcore had that whole "danger prediction" system that gave a score to all enemy players, and could pop uber/kritz depending on how many dangerous players were within view.

I can think of how to do that fairly easily, each class has a score, set a minimum danger value and if that many people are visible then pop

**Comment by johnathon-walnut** [comment]
Though I'm only really thinking of vaccinator for now

**Comment by johnathon-walnut** [comment]
No promises on implementing all ur things tho

**Comment by lnx00** [comment]
> This is god tier hacks for aimbot and other classes

Where can I find these god tier hacks you're talking about?

**Comment by johnathon-walnut** [comment]
Give me phosphor

> > This is god tier hacks for aimbot and other classes
> 
> Where can I find these god tier hacks you're talking about?

**Comment by lnx00** [comment]
> Give me phosphor
> 
> > > This is god tier hacks for aimbot and other classes
> > 
> > 
> > Where can I find these god tier hacks you're talking about?

Give me spookyballs

**Comment by SeasonalFerret** [comment]
> Give me phosphor

https://www.curseforge.com/minecraft/mc-mods/phosphor

**Comment by HutchyBen** [comment]
> > Give me phosphor
> 
> https://www.curseforge.com/minecraft/mc-mods/phosphor

Starlight is better 😡😡😡

**Comment by johnathon-walnut** [comment]
well this might never get finished but i have added a function for calculating some of the hitscan danger of an individual player, this still needs to account for sniper stuff. I won't do peek prediction on them as i imagine it would cause more false positives and a significant fps loss (assuming using gamemovement) and for huntsman arrows (inclusive of ping). still need to work on when to actually pop the uber too as at the moment it will just pop if there is any damage. so yeah i havent done much from your big list so far but maybe i will keep working on it :-D

**Comment by ghost** [comment]
gamemovement is fps friendly

**Comment by SeasonalFerret** [comment]
You only need to do peek prediction on snipers that are scoped in. I'm sure there's never more than 3 of those in any game at a time. 
And this is only to stop those bots from sniping your target.

I guess you could also run this peek prediction on snipers that are looking directly at you. But if a sniper has their yaw exactly 0 (as most bots do, since they always look straight forward), run it on them too.
if ((player = sniper) && (status = scoped)) {
if (player = looking at us) || (player_yaw = 0) {
peekDetection()
}}

**Comment by Zzzzz124** [comment]
Wow, didn't expect to drum up quite the responses, I'm really glad others share the same interest for Medic. 

A few things I'd love to see personally:
1. Auto vaccinator, the vaccinator switches to the correct resistance type based on threats and incoming bullets to the target, or myself. Auto uber would pop with vaccinator based on threats as well,  so not just switching resistances, but also popping uber in conjunction.
2. Autoheal but ON A HOTKEY, some cheats basically leverage if the user has aimbot active, and just categorizes this as an aimbot feature, the autoheal snaps on to the players in the users selected FOV, and heals the lowest ones first, and keeps snapping to the next until all players in the FOV are healed. I'd like this on a hotkey, cause I want to only do this sometimes, then switch to healing my friend. This is great for building an uber charge quick. I noticed it has to be done in a silent way, perhaps server sided, because people notice you snapping right away if not.
3. I would love to have a cham on players on your team if they get low in HP, perhaps your team color overlays the lowest people on your team. It's just an easier indicator to heal those people vs. having to keep track of health bars.
4. Let team member press X and call for Uber, then pop uber. 

Maybe some stuff I've never seen before on Medic, would be awesome. These are just features I'd love to have, and have seen before.

Thanks for your time!

**Comment by SeasonalFerret** [comment]
> I would love to have a cham on players on your team if they get low in HP, perhaps your team color overlays the lowest people on your team. It's just an easier indicator to heal those people vs. having to keep track of health bars.

ESP on low hp teammates only would be neat for figuring out who needs health, yes.

> * Let team member press X and call for Uber, then pop uber.

Not everyone presses X then reaches all the way to the number 8 on their keyboard.
Just pop uber if they call for Medic while already at full HP. This is basically them saying "get ready to uber me!"
